### PR TITLE
Add code coverage uploads to Codecov back into our CI

### PR DESCRIPTION
**Pull request by strickvl**
Restores the uploading of code coverage reports to Codecov.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: .github/codecov.yml [hunk 1]
```diff
@@ -8,9 +9,10 @@ coverage:
   status:
     project:
       default:
-        threshold: 1%   # allow coverage to drop by 1% before failing the PR
+        target: auto
+        threshold: 0%   # allow coverage to drop by 1% before failing the PR
```
Comment (avishniakov) [inline .github/codecov.yml:13]: comment is outdated

**Review by avishniakov — APPROVED** [review]
CI is failing, besides that all good from my PoV

File: .github/codecov.yml (hunk 1 above)
Comment (strickvl) [inline .github/codecov.yml:13]:
```suggestion
        threshold: 0%   # don't allow coverage to drop otherwise fail the PR
```

**Review by fa9r — COMMENTED** [review]
Not sure it makes sense to do this while the integrations are part of the coverage. As mentioned today, our codecov would be deceptively low because it includes all the non-testable integration code, and every time we'd add an integration the codecov would drop further, which would then fail the CI of the respective PR.

**Review by bcdurak — COMMENTED** [review]
Just leaving a placeholder comment, please request again once it is ready to be reviewed again.

**Pull request closed without merging**